examples/streetlights-operation-security.yml

```
@@ -0,0 +1,202 @@
+
+    * Turn a specific streetlight on/off 🌃
+    * Dim a specific streetlight 😎
+    * Receive real-time information about environmental lighting conditions 📈
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0
+
+servers:
+  test:
+    url: test.mykafkacluster.org:8092
+    protocol: kafka-secure
+    description: Test broker
+    security:
+      - saslScram: []
+  test_oauth:
+    url: test.mykafkacluster.org:8093
+    protocol: kafka-secure
+    description: Test port for oauth 
+    security:
+      - streetlights_auth:
+         - streetlights:write
+         - streetlights:read
+    
+ 
+defaultContentType: application/json
+
+channels:
+  smartylighting.streetlights.1.0.event.{streetlightId}.lighting.measured:
+    description: The topic on which measured values may be produced and consumed.
+    parameters:
+      streetlightId:
+        $ref: '#/components/parameters/streetlightId'
+    publish:
+      summary: Inform about environmental lighting conditions of a particular streetlight.
+      operationId: receiveLightMeasurement
+      traits:
+        - $ref: '#/components/operationTraits/kafka'
+      message:
+        $ref: '#/components/messages/lightMeasured'
+      security:
+       streetlights_auth:
+         - streetlights:write  
+  smartylighting.streetlights.1.0.action.{streetlightId}.turn.on:
+    parameters:
+      streetlightId:
+        $ref: '#/components/parameters/streetlightId'
+    subscribe:
+      operationId: turnOn
+      traits:
+        - $ref: '#/components/operationTraits/kafka'
+      message:
+        $ref: '#/components/messages/turnOnOff'
+      security:
+      # This operation level security implies the ability to sends a message to 
+      # `smartylighting.streetlights.1.0.action.{streetlightId}.turn.on` with Authorization headers 
+      # with `streetlights:read` scope. It is also possible for the same channel when using `test_auth` 
+      # server to instead use credentials for security (scramSha256 in this example) specified on the server level
+       streetlights_auth:
+         - streetlights:read  
+
+  smartylighting.streetlights.1.0.action.{streetlightId}.turn.off:
+    parameters:
+      streetlightId:
+        $ref: '#/components/parameters/streetlightId'
+    subscribe:
+      operationId: turnOff
+      traits:
+        - $ref: '#/components/operationTraits/kafka'
+      message:
+        $ref: '#/components/messages/turnOnOff'
+      security:
+       streetlights_auth:
+         - streetlights:read    
+  smartylighting.streetlights.1.0.action.{streetlightId}.dim:
+    parameters:
+      streetlightId:
+        $ref: '#/components/parameters/streetlightId'
+    subscribe:
+      operationId: dimLight
+      traits:
+        - $ref: '#/components/operationTraits/kafka'
+      message:
+        $ref: '#/components/messages/dimLight'
+      security:
+      # This operation level security implies the ability to sends a message to 
```

```suggestion
      # This operation level security implies the ability to send a message to 
```

cc @alequetzalli in case you want to review this comment (it is a public example).